tests/test_utils.py

```diff
@@ -1083,11 +1097,10 @@ def test_index_has_overlap(obj, expected):
     ]
 )
 def test_intersect(objs, expected):
-    for permuted_objs in itertools.permutations(objs):
```

Comment:
In `test_symmetric_difference()` we still have the permutation tests, but sort the results:

```python
    expected = expected.sortlevel()[0]
    for permuted_objs in itertools.permutations(objs):
        pd.testing.assert_index_equal(
            audformat.utils.symmetric_difference(permuted_objs).sortlevel()[0],
            expected,
        )
```

Maybe add it here as well?

Reply:
Unfortunately, sorting is not enough as we have the problem discussed above that the returned index type depends on the order as well, which means we would get errors like this here:

```
E       [left]:  Int64Index([1], dtype='int64', name='idx')
E       [right]: Index([1], dtype='Int64', name='idx')
```

Reply:
The other problem with sorting the results is that we cannot test for the desired order. At the moment I added the following test:

```python
        (
            [
                audformat.filewise_index(['f3', 'f2', 'f1']),
                audformat.filewise_index(['f1', 'f2']),
            ],
            audformat.filewise_index(['f2', 'f1']),
        ),
```
which would be meaningless if we use sort.

Reply:
In `test_symmetric_difference()` we first test that the result equals the expected result and after that we do the permutation tests. But I see your point with the changing `dtype`. One way to have a least deterministic `dtypes` would be to always use `Int64` and `boolean` in `set_index_dtypes()`.

Reply:
We could also update `_alike_index()` instead if we want to allow `set_index_dtypes()` to also return `int64`, which might be desirable in some cases.

Reply:
Sorry, I was not precise and meant not to change `set_index_dtypes()` but `dtypes` before we pass it to `set_index_dtypes()`.

Reply:
But maybe it's also fine to simply state in the docstring that the results will have the same `dtypes` as the first object.

Reply:
I added a statement to the docstring.